File: stack/Solution20.java
import java.util.Stack;

class Solution20 {
    public boolean isValid(String s) {
    	Stack<Integer> stack = new Stack<>();
    	for(int i = 0; i < s.length(); i++){
    		switch(s.charAt(i)){
    			case '(': 
    			stack.push(1);
    			break;

    			case '[':
    			stack.push(2);
    			break;

    			case '{':
    			stack.push(3);
    			break;

    			case ')':
    			if(stack.empty()){
    				return false;
    			}
    			if(stack.pop() != 1){
    				return false;
    			}
    			break;

    			case ']':
    			if(stack.empty()){
    				return false;
    			}
    			if(stack.pop() != 2){
    				return false;
    			}
    			break;

    			case '}':
    			if(stack.empty()){
    				return false;
    			}
    			if(stack.pop() != 3){
    				return false;
    			}
    			break;

    			default:
    			return false;
    		}
    	}
    	return stack.empty();
    }
}

File: C++/Solution20.cpp
#include<iostream>
#include<vector>

#include<stack>

using namespace std;

class Solution20 {
public:
    bool isValid(string s) {
        stack<char> stk;
        for(int i = 0; i < s.length(); i++){
            char cur = s[i];
            if(cur == '(' || cur == '[' || cur == '{'){
                stk.push(cur);
            }
            else{
                if(cur == ')'){
                    if(stk.empty() || stk.top() != '('){
                        return false;
                    }
                    stk.pop();
                }
                if(cur == ']'){
                    if(stk.empty() || stk.top() != '['){
                        return false;
                    }
                    stk.pop();
                }
                if(cur == '}'){
                    if(stk.empty() || stk.top() != '{'){
                        return false;
                    }
                    stk.pop();
                }
            }
        }
        return stk.empty();        
    }
};